api/src/db/models/user_models.py

```diff
@@ -82,3 +86,18 @@ class UserSavedOpportunity(ApiSchemaTable, TimestampMixin):
     opportunity: Mapped[Opportunity] = relationship(
         "Opportunity", back_populates="saved_opportunities_by_users"
     )
+
+
+class UserSavedSearch(ApiSchemaTable, TimestampMixin):
+    """Table for storing saved search queries for users"""
+
+    __tablename__ = "user_saved_search"
+
+    saved_search_id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True, default=uuid.uuid4)
+
+    user_id: Mapped[uuid.UUID] = mapped_column(ForeignKey(User.user_id), index=True)
+    user: Mapped[User] = relationship(User, back_populates="saved_searches")
+
+    search_query: Mapped[dict] = mapped_column(JSONB, nullable=False)
+
+    name: Mapped[str] = mapped_column(nullable=False)
```

Comment:
Can skip out on the nullable bit, SQLAlchemy gets that from the python typing.

```suggestion
    search_query: Mapped[dict] = mapped_column(JSONB)

    name: Mapped[str]
```